org.eclipse.scanning.event.ui/src/org/eclipse/scanning/event/ui/view/StatusQueueView.java

```diff
@@ -485,42 +478,120 @@ protected void togglePausedConsumer(IAction pauseConsumer) {
 			pauser.broadcast(pbean);
 
 		} catch (Exception e) {
-			ErrorDialog.openError(getViewSite().getShell(), "Cannot pause queue "+getSubmissionQueueName(), "Cannot pause queue "+getSubmissionQueueName()+"\n\nPlease contact your support representative.",
-					new Status(IStatus.ERROR, Activator.PLUGIN_ID, e.getMessage()));
+			ErrorDialog.openError(getViewSite().getShell(), "Cannot pause queue "+getSubmissionQueueName(),
+				"Cannot pause queue "+getSubmissionQueueName()+"\n\nPlease contact your support representative.",
+				new Status(IStatus.ERROR, Activator.PLUGIN_ID, e.getMessage()));
 		}
 		pauseConsumer.setChecked(queueConnection.isQueuePaused(getSubmissionQueueName()));
 	}
 
-	protected void pauseJob() {
+	private void pauseActionUpdate(Action pause, List<StatusBean> statusBeans, String description) {
+		boolean allStatusBeansArePaused = statusBeans.stream().allMatch(sb -> sb.getStatus().isPaused());
+		boolean noStatusBeansArePaused = statusBeans.stream().noneMatch(sb -> sb.getStatus().isPaused());
+
+		// Only enable the pause button if jobs are active and they are all in the same state
+		pause.setEnabled(allStatusBeansArePaused ^ noStatusBeansArePaused); // XOR
+		pause.setChecked(allStatusBeansArePaused);
+		pause.setText((allStatusBeansArePaused ? "Resume" : (
+				noStatusBeansArePaused ? "Pause" : "Mixed paused and unpaused" ))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3358)